resources/lib/nodefunctions.py

```diff
@@ -435,6 +435,115 @@ def extractID( self, path ):
         return itemID
 
+# ##############################################
+# ### Functions to externally set properties ###
+# ##############################################
+
+    def setProperties( self, properties, values, labelID, group, DATA ):
+        # This function will take a list of properties and values and apply them to the
+        # main menu item with the given labelID
+        if not group:
+            group = "mainmenu"
+
+        # Decode values
+        properties = try_decode( properties )
+        values = try_decode( values )
+        labelID = try_decode( labelID )
+        group = try_decode( group )
+
+        # Split up property names and values
+        propertyNames = properties.split( "|" )
+        propertyValues = values.split( "|" )
+        labelIDValues = labelID.split( "|" )
+        if len( labelID ) == 0:
+            # No labelID passed in, lets assume we were called in error
+            return
+        if len( propertyNames ) == 0:
+            # No values passed in, lets assume we were called in error
+            return
+
+        # Get user confirmation that they want to make these changes
+        message = "Set %s property to %s?" %( propertyNames[ 0 ], propertyValues[ 0 ] )
+        if len( propertyNames ) == 2:
+            message += "[CR](and 1 other property)"
+        elif len( propertyNames ) > 2:
+            message += "[CR](and %d other properties)" %( len( propertyNames ) -1 )
+        shouldRun = xbmcgui.Dialog().yesno( __addon__.getAddonInfo( "name" ), message )
+        if not shouldRun:
+            return
+
+        # Load the properties
+        currentProperties, defaultProperties = DATA._get_additionalproperties( xbmc.translatePath( "special://profile/" ).decode( "utf-8" ) )
+        otherProperties, requires, templateOnly = DATA._getPropertyRequires()
+
+        # If there aren't any currentProperties, use the defaultProperties instead
+        if currentProperties == [None]:
+            currentProperties = defaultProperties
+
+        # Pull out all properties into multi-dimensional dicts
+        allProps = {}
+        allProps[ group ] = {}
+        for currentProperty in currentProperties:
+            # If the group isn't in allProps, add it
+            if currentProperty[ 0 ] not in allProps.keys():
+                allProps[ currentProperty [ 0 ] ] = {}
+            # If the labelID isn't in the allProps[ group ], add it
+            if currentProperty[ 1 ] not in allProps[ currentProperty[ 0 ] ].keys():
+                allProps[ currentProperty[ 0 ] ][ currentProperty[ 1 ] ] = {}
+            # And add the property to allProps[ group ][ labelID ]
+            if currentProperty[ 3 ] is not None:
+                allProps[ currentProperty[ 0 ] ][ currentProperty [ 1 ] ][ currentProperty[ 2 ] ] = currentProperty[ 3 ]
+
+        # Loop through the properties we've been asked to set
+        count = -1
+        for propertyName in propertyNames:
```

Comment:
for count, propertyName in enumerate(propertyNames)
would be the pythonic way for any kind of loop with a counter. That way you dont have to initalize and increment manually.